src/main.cpp

```diff
@@ -44,13 +36,50 @@ class input_t {
 	std::set<key_t> m_held;
 };
 
-void tick(world_clock_t& out_clock, input_t const& input, sec_t dt) noexcept {
-	static constexpr f32 scale = 10.0f;
-	f32 delta{};
-	if (input.any_held(key_t::Down, key_t::Left, key_t::S, key_t::A)) { delta += -1.0f; }
-	if (input.any_held(key_t::Up, key_t::Right, key_t::W, key_t::D)) { delta += 1.0f; }
-	out_clock += scale * delta * dt.count();
-}
+class clock_ticker_t {
+  public:
+	clock_ticker_t(input_t const* input, world_clock_t* clock) noexcept : m_input(input), m_clock(clock) {}
+
+	void operator()(sec_t dt) noexcept {
+		if (m_input->any_held(key_t::Enter, key_t::R)) { update(flag::zeroing); }
+		static constexpr f32 translation_scale = 10.0f;
+		if (test(flag::zeroing)) {
```

Comment:
Gist of this branch: keep subtracting `delta` until less than that remains, and then reset the flag.